R/layers.R

```diff
@@ -177,24 +177,125 @@ popupOptions = function(
   )
 }
 
-#' @param icon the icon for markers; if you want to create a new icon using
-#'   JavaScript, please remember to use \code{\link[htmlwidgets]{JS}()} on the
-#'   JavaScript string; see \url{http://leafletjs.com/reference.html#icon}
+#' @param icon the icon(s) for markers; you can either use literal JavaScript
+#'   code (e.g. \code{\link[htmlwidgets]{JS}('L.icon({iconUrl: '?', size: [?,
+#'   ?]})')} or an R list to create an icon (e.g. \code{list(iconUrl = '?', size
+#'   = c(x, y))}), or use \code{\link{iconList}()} to create a list of icons;
```

Comment:
I think it's a bit confusing to allow either js or R code. Why not have a small helper function like `leaftlet_icon <- function(url, width, height) {}`.  Then instead of `iconList()` you'd just use a (named) list of icons